File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int dest = nums.length - 1;

        for(int i = nums.length - 1 ; i > 0 ; i--)
        {
            if(nums[i - 1] + (i - 1) >= dest)
            {
                dest = i - 1;
            }
        }

        return dest == 0;
    }
}

File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxReach = 0;

        // 在所能及范围内遍历
        for(int i = 0 ; i <= maxReach ; i++){
            maxReach = max(maxReach, i + nums[i]); // 更新最远可以跳到的位置
            if(maxReach >= nums.size() - 1) return true;
        }

        return false;
    }
};